pkg/kubelet/cm/devicemanager/plugin/v1beta1/server.go

```diff
@@ -190,3 +200,20 @@ func (s *server) visitClients(visit func(r string, c Client)) {
 	}
 	s.mutex.Unlock()
 }
+
+func (s *server) Name() string {
+	return "device plugin registration server"
+}
+
+func (s *server) Check(_ *http.Request) error {
```

Comment:
can you remind me why it needs the http.Request as an argument? It would be great to eliminate this dependency

Reply:
Because the `HealthChecker` interface is used in the kubelet health checks (/healthz ) as well as in the watchdog.

```
// HealthChecker is a named healthz checker.
type HealthChecker interface {
	Name() string
	Check(req *http.Request) error
}
```

Reply:
yes, I understand that interface has it. The question is whether any checked is actually using the `req` for anything

Reply:
1. It appears that Kubernetes internally does not use this parameter, but there are many implementations of this interface. If we need to remove this parameter, it might be better to do so in a new PR. 
2. This is because the interface is defined in `k8s.io/apiserver`, and it is uncertain whether out-of-tree or custom services defined by K8s users might be using this interface.